### ISSUE TITLE
Interpolation in options.0.option_settings.value causes error

### ISSUE THREAD

**Issue by eangeeag** [terraform-aws-modules/terraform-aws-rds#142]
See comment [here](https://github.com/terraform-aws-modules/terraform-aws-rds/issues/63#issuecomment-508524071)

Passing an interpolated value to the `value` field of `option_settings` causes an error:
```
options = [
    {
      option_name = "SQLSERVER_BACKUP_RESTORE"

      option_settings = [
        {
          name  = "IAM_ROLE_ARN"
          value = "${module.iam.rds_s3_integration_role_arn}"
        },
      ]
    },
  ]
```

Error: `module.bike_db.module.db_option_group.aws_db_option_group.this: "option.0.option_name": required field is not set`

Passing a hardcoded ARN removes the problem.

**Comment by horse80** [comment]
Can confirm this in 0.11, I also have this issue. Setting a fix ARN would solve this, but of course i need to use interpolation. 

So this works:
```
  options = [
    {
      option_name = "SQLSERVER_BACKUP_RESTORE"

      option_settings = [
        {
          name  = "IAM_ROLE_ARN"
          value = "hardcoded ARN"
        },
      ]
    },
  ]
```

This not:
```
  options = [
    {
      option_name = "SQLSERVER_BACKUP_RESTORE"

      option_settings = [
        {
          name  = "IAM_ROLE_ARN"
          value = "${aws_iam_role.my_iam_role.arn}"
        },
      ]
    },
  ]
```

**Comment by horse80** [comment]
Forgot to mention the context, sorry.
So, i am using this within a module referencing "terraform-aws-modules/rds/aws" as source.

The only way i can work this around is that i create an _aws_db_option_group_ resource, and set the options within, then i reference this _aws_db_option_group_ instance.
```
resource "aws_db_option_group" "rds_op_grp" {
  name                     = "rds_op_grp"
  option_group_description = "Terraform Option Group"
  engine_name              = "${local.rds_engine}"
  major_engine_version     = "${local.major_engine_version}"

  option {
    option_name = "SQLSERVER_BACKUP_RESTORE"

    option_settings {
      name  = "IAM_ROLE_ARN"
      value = "${aws_iam_role.my_iam_role.arn}"
    }
  }

  tags = "${var.tags}"
}
```

**Comment by bryantbiggs** [comment]
hi all, this should be resolved on later versions of Terraform - 0.11.x is no longer supported by the module
